src/lib/fetch.ts

```diff
@@ -23,17 +23,22 @@ export const resolveHeadersConstructor = () => {
 }
 
 export const fetchWithAuth = (
+  supabaseAnonKey: string,
   getAccessToken: () => Promise<string | null>,
   customFetch?: Fetch
 ): Fetch => {
   const fetch = resolveFetch(customFetch)
   const HeadersConstructor = resolveHeadersConstructor()
 
   return async (input, init) => {
-    const accessToken = await getAccessToken()
+    const accessToken = (await getAccessToken()) ?? supabaseAnonKey
     let headers = new HeadersConstructor(init?.headers)
 
-    if (!headers.has('Authorization') && accessToken) {
+    if (!headers.has('apikey')) {
+      headers.append('apikey', supabaseAnonKey)
```

Comment:
Should this be `.set` instead of `.append`? Also we can set this unconditionally since this is only used for Kong

Reply:
Nice catch - I'll blame copilot for that one ;)

Reply:
I think the conditional is nice in case you're using the the anon key and override with the service role